sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -21,75 +24,109 @@ namespace usm {
 
 void *alignedAlloc(size_t Alignment, size_t Size, const context &Ctxt,
                    alloc Kind) {
-  std::shared_ptr<context_impl> CtxImpl = detail::getSyclObjImpl(Ctxt);
-  std::shared_ptr<USMDispatcher> Dispatch = CtxImpl->getUSMDispatch();
-  pi_context C = CtxImpl->getHandleRef();
-  pi_result Error;
   void *RetVal = nullptr;
-
-  switch (Kind) {
-  case alloc::host: {
-    RetVal = Dispatch->hostMemAlloc(C, nullptr, Size, Alignment, &Error);
-    break;
-  }
-  case alloc::device:
-  case alloc::shared:
-  case alloc::unknown: {
-    RetVal = nullptr;
-    Error = PI_INVALID_VALUE;
-    break;
-  }
+  if (Ctxt.is_host()) {
+    if (Alignment) {
+      aligned_allocator<char> Alloc(Alignment);
+      try {
+        RetVal = Alloc.allocate(Size);
+      } catch (const std::bad_alloc &) {
+        // Conform with Specification behavior
+        RetVal = nullptr;
+      }
+    } else {
+      RetVal = ::malloc(Size);
```

Comment:
Just out of curiosity: is there any benefits in using `malloc` instead of `aligned_allocator` if `Alignment` is not set? Can we use `aligned_allocator` with some default alignment instead of malloc (e.g. default constructor of `aligned_allocator`)?

Reply:
better match normal system behavior?